benchmarks/src/test/java/io/crate/execution/engine/aggregation/GroupingLongCollectorBenchmark.java

```diff
@@ -130,4 +142,22 @@ public void measureGroupBySumLongOptimized(Blackhole blackhole) throws Exception
         rowsIterator = InMemoryBatchIterator.of(rows, SENTINEL);
         blackhole.consume(BatchIterators.collect(rowsIterator, groupBySumSingleLongCollector).get());
     }
+
+    /**
+     * To establish a base line on how fast it could go
+     */
+    @Benchmark
+    public LongObjectHashMap<Long> measureGroupingOnLongArray() {
+        LongObjectHashMap<Long> sumByKey = new LongObjectHashMap<>();
+        for (long number : numbers) {
+            sumByKey.compute(number, (key, oldValue) -> {
+                if (oldValue == null) {
+                    return 0L;
```

Comment:
should this be `number`? (adding the first value for a key right? )

Reply:
right, pushed a fixup.